packages/typespec-test/test/unionwidget_dpg/spec/main.tsp

```diff
@@ -0,0 +1,257 @@
+  fooProp: string;
+}
+
+model WidgetData3 {
+  @projectedName("json", "bar_prop")
+  barProp: string;
+}
+
+model WidgetData4 {
+  start: utcDateTime;
+  end?: utcDateTime;
+}
+
+model WidgetData5 {
+  data: bytes;
+}
+
+model WidgetData6 {
+  data: WidgetData5;
+}
+
+model Widget {
+  @key id: string;
+  weight: int32;
+  color: "red" | "blue";
+}
+
+model Widget1 extends Widget {
+  data: WidgetData0 | WidgetData1
+}
+
+model Widget2 extends Widget {
+  data: WidgetData0 | WidgetData2
+}
+
+model Widget3 extends Widget {
+  data: WidgetData1 | WidgetData2
+}
+
+model Widget4 extends Widget {
+  data: WidgetData2 | WidgetData3
+}
+
+model Widget5 extends Widget {
+  data: WidgetData0 | WidgetData4
+}
+
+model Widget6 extends Widget {
+  data: WidgetData0 | WidgetData5
+}
+
+model Widget7 extends Widget {
+  data: WidgetData0[] | WidgetData1[]
+}
+
+model Widget8 extends Widget {
+  data: WidgetData0[] | WidgetData2[]
+}
+
+model Widget9 extends Widget {
+  data: WidgetData1[] | WidgetData2[]
+}
+
+model Widget10 extends Widget {
+  data: WidgetData2[] | WidgetData3[]
+}
+
+model Widget11 extends Widget {
+  data: WidgetData0[] | WidgetData4[]
+}
+
+model Widget12 extends Widget {
+  data: WidgetData0[] | WidgetData5[]
+}
+
+model Widget13 extends Widget {
+  data: WidgetData0[] | WidgetData1;
+}
+
+model Widget14 extends Widget {
+  data: WidgetData0 | WidgetData1[];
+}
+
+model Widget15 extends Widget {
+  data: WidgetData0[] | WidgetData2;
+}
+
+model Widget16 extends Widget {
+  data: WidgetData0 | WidgetData2[];
+}
+
+model Widget17 extends Widget {
+  data: WidgetData1[] | WidgetData2;
+}
+
+model Widget18 extends Widget {
+  data: WidgetData1 | WidgetData2[];
+}
+
+model Widget19 extends Widget {
+  data: WidgetData2[] | WidgetData3;
+}
+
+model Widget20 extends Widget {
+  data: WidgetData2 | WidgetData3[];
+}
+
+model Widget21 extends Widget {
+  data: WidgetData0[] | WidgetData4
+}
+
+model Widget22 extends Widget {
+  data: WidgetData0 | WidgetData4[]
+}
+
+model Widget23 extends Widget {
+  data: WidgetData0[] | WidgetData5
+}
+
+model Widget24 extends Widget {
+  data: WidgetData0 | WidgetData5[]
+}
+
+model Widget25 extends Widget {
+  data: WidgetData0 | WidgetData3 | WidgetData5[]
+}
+
+model Widget26 extends Widget {
+  data: WidgetData2 | WidgetData4 | WidgetData6[]
+}
+
+model Widget27 extends Widget {
+  data: WidgetData0 | utcDateTime;
+}
+
+model Widget28 extends Widget {
+  data: WidgetData0 | bytes;
+}
+
+model Widget29 extends Widget {
+  data: utcDateTime | string;
+}
+
+model Widget30 extends Widget {
+  data: bytes | string;
+}
+
+model Widget31 extends Widget {
+  data: utcDateTime | bytes;
+}
+
+model Widget32 extends Widget {
+  data: WidgetData0 | Record<string>;
+}
+
+model Widget33 extends Widget {
+  data: WidgetData2 | Record<string>;
+}
+
+@error
+model Error {
+  code: int32;
+  message: string;
+}
+
+interface WidgetService {
+  @get @route("customGet/response") customGetResponse(): Widget;
```

Comment:
will we have any difference if the union happens in body and in normal property e.g binary type?